plugins/apollo-explorer/src/components/ApolloExplorerPage/ApolloExplorerPage.tsx

```diff
@@ -35,19 +41,37 @@ type EndpointProps = {
   };
 };
 
+/**
+ * Export types to be used with {@link @backstage/apollo-explorer#ApolloExplorerPage}.
+ *
+ * @public
+ */
+export type AuthCallback = (options: {
+  apiHolder: ApiHolder;
+}) => Promise<string>;
+
 type Props = {
   title?: string | undefined;
   subtitle?: string | undefined;
   endpoints: EndpointProps[];
+  authCallback?: AuthCallback;
```

Comment:
This needs to be configured as part of the endpoint right? Seems pretty reasonable that you might need different auth flows for different endpoints. Or perhaps a middle ground to keep things simple is that we forward the `endpointUrl` to the auth callback.

Reply:
[3f422fb](https://github.com/backstage/backstage/pull/23405/commits/3f422fbd66bc0262d9df60c7f8c5a6ca5fa182c0) should cover this